lib/spack/docs/build_settings.rst

```diff
@@ -57,10 +57,13 @@ directory. Here's an example of an external configuration:
 
    packages:
      openmpi:
-       paths:
-         openmpi@1.4.3%gcc@4.4.7 arch=linux-debian7-x86_64: /opt/openmpi-1.4.3
-         openmpi@1.4.3%gcc@4.4.7 arch=linux-debian7-x86_64+debug: /opt/openmpi-1.4.3-debug
-         openmpi@1.6.5%intel@10.1 arch=linux-debian7-x86_64: /opt/openmpi-1.6.5-intel
+       externals:
+       - spec: "openmpi@1.4.3%gcc@4.4.7 arch=linux-debian7-x86_64"
+         prefix: /opt/openmpi-1.4.3
+       - spec: "openmpi@1.4.3%gcc@4.4.7 arch=linux-debian7-x86_64+debug"
+         prefix: /opt/openmpi-1.4.3-debug
+       - spec: "openmpi@1.6.5%intel@10.1 arch=linux-debian7-x86_64"
+         prefix: /opt/openmpi-1.6.5-intel
```

Comment:
I definitely like this syntax a lot better. I've seen complaints about having to specify the package name multiple times. I wonder if we could move externals up one level, or not require the package name in `spec`.

Reply:
Also, another common request we get for external packages is the ability to specify where headers/libs are, so this new syntax allows one to add additional attributes really easily.

Reply:
> I wonder if we could move externals up one level, or not require the package name in spec.

Given that most of these entries will be automatically generated in the long run I don't see repeating a name as a big issue at the moment.

> Also, another common request we get for external packages is the ability to specify where headers/libs are, so this new syntax allows one to add additional attributes really easily.

This was exactly a design goal of the PR. The rationale is that we want to prepare stage for all the attributes that are now in `compilers.yaml` to be moved to `packages.yaml`. This PR permits on a package by package base to add extra custom attributes to `packages.yaml`.